app/lib/common/MarketClasses.js

```diff
@@ -721,12 +733,92 @@ class CallOrder {
     getCollateral() {
         if (this._collateral) return this._collateral;
         return (this._collateral = new Asset({
-            amount: this.for_sale,
-            asset_id: this.for_sale_id,
-            precision: this.assets[this.for_sale_id].precision
+            amount: this.collateral,
+            asset_id: this.collateral_id,
+            precision: this.assets[this.collateral_id].precision
         }));
     }
 
+    _getMaxCollateralToSell() {
+        /*
+        BSIP38:
+        * max_amount_to_sell = (debt * target_CR - collateral * feed_price)
+        * / (target_CR * match_price - feed_price)
+        */
+        if (this.target_collateral_ratio) {
+            let feed_price = this._getFeedPrice();
+            let match_price = this._getMatchPrice();
+            let nominator =
+                this.debt * this.target_collateral_ratio -
```

Comment:
Make sure `target_collateral_ratio` here (and same below) has been divided by 1000, for example, when the value is `1750`, need to use `1.75`.

Reply:
Right I remember wondering if I had to do that. Right now it's being used as the full int value, 1750, and all amounts are in satoshis. The prices are also adjusted to be useable with satoshis in _getFeedPrice and _getMatchPrice.

So even when using satoshi amounts the ratio should be divided by 1000?

Reply:
Yes. Or multiply the other number by 1000, but also need to do multiplication for the denominator.

Reply:
Is the number supposed to be negative?